### ISSUE TITLE
Bad example - 100% CPU due to looping

### ISSUE THREAD

**Issue by regulatre** [wadda/gps3#26]
The sample code below is currently returned as the top search result for python gps. It's hosted at this URL https://pypi.org/project/gps3/  and the big problem here is that the loop runs out of control, trashing the CPU usage for no reason. I would recommend adding a sleep to the loop. Quite simply a "time.sleep(0.2)" just below the "for new_data in ..." line would save a ton of carbon as N00bs using the sample code don't end up with code that max's out their CPU with useless looping. 

```
from gps3 import gps3
gps_socket = gps3.GPSDSocket()
data_stream = gps3.DataStream()
gps_socket.connect()
gps_socket.watch()
for new_data in gps_socket:
    if new_data:
        data_stream.unpack(new_data)
        print('Altitude = ', data_stream.TPV['alt'])
        print('Latitude = ', data_stream.TPV['lat'])
```

**Comment by socieboy** [comment]
@regulatre I don't think this repo is even maintained right now.... i'm wonder if you find something else to work with?

**Comment by tfeldmann** [comment]
I've written an alternative client because of this. Uses nearly no CPU and some modern best practices.
https://github.com/tfeldmann/gpsdclient
